beacon-chain/state/state-native/getters_withdrawal.go

```diff
@@ -183,6 +183,15 @@ func (b *BeaconState) ExpectedWithdrawals() ([]*enginev1.Withdrawal, uint64, err
 	return withdrawals, partialWithdrawalsCount, nil
 }
 
+func (b *BeaconState) PendingPartialWithdrawals() ([]*ethpb.PendingPartialWithdrawal, error) {
+	if b.version < version.Electra {
+		return nil, errNotSupported("PendingPartialWithdrawals", b.version)
+	}
+	b.lock.RLock()
```

Comment:
shouldn't we have this at the top even if we are just reading version?

Reply:
Version doesn't change for the lifetime of the object, therefore there is no point in locking it